tap-version-14-specification.md

```diff
@@ -0,0 +1,1167 @@
+# todo: false
+ok 2 - hello \# todo
+
+# description: hello
+# todo: true
+# todo reason: hash # character
+ok 3 - hello # todo hash \# character
+# (assuming "character" isn't a known custom directive)
+ok 4 - hello # todo hash # character
+
+# description: hello \
+# todo: true
+# todo reason: hash # character
+ok 5 - hello \\# todo hash \# character
+# (assuming "character" isn't a known custom directive)
+ok 6 - hello \\# todo hash # character
+
+# description: hello # description # todo
+# todo: false
+# (assuming "description" isn't a known custom directive)
+ok 7 - hello # description # todo
+
+# multiple escaped \ can appear in a row
+# description: hello \\\# todo
+# todo: false
+ok 8 - hello \\\\\\\# todo
+
+1..8
+```
+
+## Subtests
+
+Subtests provide a way to nest one TAP14 stream inside another.  This is
+useful in a variety of situations.  For example:
+
+1. A Harness parses a collection of TAP documents, providing output which
+   is also in TAP format.
+
+    ```tap
+    TAP version 14
+    1..2
+
+    # Subtest: foo.tap
+        1..2
+        ok 1
+        ok 2 - this passed
+    ok 1 - foo.tap
+
+    # Subtest: bar.tap
+        ok 1 - object should be a Bar
+        not ok 2 - object.isBar should return true
+          ---
+          found: false
+          wanted: true
+          at:
+            file: test/bar.ts
+            line: 43
+            column: 8
+          ...
+        ok 3 - object can bar bears # TODO
+        1..3
+    not ok 2 - bar.tap
+      ---
+      fail: 1
+      todo: 1
+      ...
+    ```
+
+2. A test framework Producer provides an API for grouping assertions about
+   a related subject.  For example:
+
+    ```js
+    import t from 'tap'
+    t.ok(true, 'true is ok')
+    t.test('this is a subtest', subtest => {
+      subtest.pass('this is fine')
+      subtest.fail('this is not fine')
+      subtest.end()
+    })
+    ```
+
+    which produces the output:
+
+    ```tap
+    TAP version 14
+    ok 1 - true is ok
+    # Subtest: this is a subtest
+        ok 1 - this is fine
+        not ok 2 - this is not fine
+        1..2
+    not ok 2 - this is a subtest
+    1..2
+    ```
+
+Subtests are designed with graceful fallback for TAP 13 harnesses in mind.
+
+Since TAP 13 specifies that non-TAP output _should_ be ignored or provided
+directly to the user, and indented Test Points and Plans are non-TAP
+according to TAP 13, only the terminating correlated test point will be
+interpreted by most TAP 13 Harnesses.  Thus, they will usually report the
+overall subtest result correctly, even if they lack the details about the
+results of the subtest.
+
+Since several TAP 13 parsers in popular usage treat a repeated Version
+declaration as an error, even if the Version is indented, Subtests _should
+not_ include a Version, if TAP 13 Harness compatibility is desirable.
+
+### Bare Subtests
+
+In its simplest form, a Subtest is introduced by a 4-space indented Test
+Point, Version, Plan or Pragma line.
```

Comment:
>or Pragma line.

Maybe add `(when the harness supports it)`, or something similar? I think it is implicit as before we say that pragmas are optional for implementers, but :man_shrugging: